android/src/com/parse/parsedevday/view/TalkListFragment.java

```diff
@@ -73,12 +80,45 @@ public void done(List<Talk> talks, ParseException e) {
          * Add all of the talks to the adapter, skipping any that weren't favorited, if
          * favoritesOnly is true.
          */
-                for (Talk talk : talks) {
-                    if (!favoritesOnly || talk.isAlwaysFavorite() || Favorites.get().contains(talk)) {
-                        adapter.add(talk);
+                if (nearbyRoomsOnly) {
+                    Set<Talk> talksInRoomsWithFoundIBeacons = getTalksByFoundIBeacons(talks);
+                    adapter.addAll(talksInRoomsWithFoundIBeacons);
+
+                    setOnEmptyBeaconsListView();
```

Comment:
Didn't get why you need this function? And why do you use it only in case of nearbyRoomsOnly=true .